### ISSUE TITLE
This doesn't find *all* solutions, right?

### ISSUE THREAD

**Issue by reckoner** [tclose/Diophantine#1]
Also, I noticed that it is unable to find zero-valued solutions. Am I missing something?
Thanks!

**Comment by RichardWeissssssss** [comment]
Yes, it appears that with homogeneous systems only the trivial solution is being calculated.

When I execute `solve(A, b)` where `A = P - I`, `P = sp.Matrix([[0, 1, 0], [1, 0, 0], [1, 1, 0]])`, `I = sp.eye(3)` and `b = sp.zeros(3, 1)`, then I only get the trivial solution. When running `sp.linsolve((A, b))`, one can clearly see, that this is not the only one.
